app/src/main/java/org/mozilla/fenix/settings/wallpaper/WallpaperSettingsFragment.kt

```diff
@@ -49,6 +54,11 @@ class WallpaperSettingsFragment : Fragment() {
                             wallpaperManager.currentWallpaper = selectedWallpaper
                         },
                         onViewWallpaper = { findNavController().navigate(R.id.homeFragment) },
+                        tapLogoSwitchChecked = wallpapersSwitchedByLogo,
+                        tapLogoSwitchOnCheckedChange = {
+                            settings.wallpapersSwitchedByLogoTap = it
+                            wallpapersSwitchedByLogo = it
```

Comment:
Nice ✨  this what was causing me some Headaches, as for me the switch was not updating after click 👍🏽

Reply:
I guess compose it's smart enough to just re-render only the changed bits, right?  As `wallpapersSwitchedByLogo` is parameter of `WallpaperSettings`.

Reply:
There's a few important bits to make this work. Lemme deconstruct this line a little bit:
`var wallpapersSwitchedByLogo by remember { mutableStateOf(settings.wallpapersSwitchedByLogoTap) }`

`mutableStateOf` changes the underlying type of `wallpapersSwitchedByLogo` to a `State<Boolean>` When `State` variables are read from within a composition, the composition will automatically subscribe to changes to the variable. This means that when `wallpapersSwitchedByLogo` is updated, the composition can recompose the elements of the UI tree that use `wallpapersSwitchedByLogo`. The runtime does this in what they're calling "smart recomposition", which means that only `@Composable` functions that use the variable should get redrawn. 

`by remember { }` is a delegate that does two things: 
- lets us treat the variable as a `Boolean` instead of a `State<Boolean>`
- informs the composition that this value does not need to be initialized again when the composable is called again, which effectively gives the composable memory

Reply:
Thanks for the clarification 👍🏽